src/app/modules/form/form.service.ts

```diff
@@ -112,6 +112,39 @@ export const retrieveFullFormById = (
   })
 }
 
+/**
+ * Retrieves the specified form fields of the given formId
+ * @param formId the id of the form
+ * @param fields an array of field names to retrieve
+ * @returns ok(form) if form exists
+ * @returns err(FormNotFoundError) if the form or form admin does not exist
+ * @returns err(DatabaseError) if error occurs whilst querying the database
+ */
+export const retrieveFormFieldsById = (
```

Comment:
what's the purpose of this function? doesn't seem to be called anywhere

Reply:
if we remove it then it seems we can remove a bunch of other changes also

Reply:
'twas implementation that I did to facilitate partial retrieval of fields within a form, to conserve bandwidth between app and data store. While not actually used in this PR, I feel that you would find this helpful for subsequent PRs for the API refactor.

Reply:
I see, in that case might it be better to call it `retrieveFormKeysById`, rather than `FormFields`? I was confused because when I see "form fields" I think form fields as in fields to fill in. obviously this is subjective so I'm ok either way